src/Frame.h

```diff
@@ -218,6 +218,29 @@ class Frame_REQUEST_CHANNEL : public Frame_REQUEST_Base {
       : Frame_REQUEST_CHANNEL(streamId, flags, 0, std::move(payload)) {}
 };
 
+class Frame_REQUEST_RESPONSE {
+ public:
+  static constexpr bool Trait_CarriesAllowance = false;
+
+  Frame_REQUEST_RESPONSE() = default;
+  Frame_REQUEST_RESPONSE(StreamId streamId, FrameFlags flags, Payload payload)
+      : header_(
+            FrameType::REQUEST_RESPONSE,
+            flags | payload.getFlags(),
+            streamId),
+        payload_(std::move(payload)) {
+    payload_.checkFlags(header_.flags_); // to verify the client didn't set
```

Comment:
This comment points out that we should probably set metadata flag completely based on the presence or not.  Generally this is an existing problem, but REQUEST_FNF also does "flags | payload.getFlags()" while other frame types don't.